### ISSUE TITLE
[IMPROVEMENT] Separate the default backup target settings from default settings

### ISSUE THREAD

**Issue by derekbit** [longhorn/longhorn#10089]
## Is your improvement request related to a feature? Please describe (👍 if you like this request)

The default backup target settings are removed. To prevent any confusion, we need to decouple these settings from global settings in chart `values.yaml` when working on a PR on longhorn repo.

**Comment by derekbit** [comment]
We can move the three settings from [chart/values.yaml#L196](https://github.com/longhorn/longhorn/blob/master/chart/values.yaml#L196) into **a new category**, such as default-backupstore. However, the rendered manifest generated from the Helm values will still be included in default-settings.yaml ([source code](https://github.com/longhorn/longhorn/blob/master/chart/templates/default-setting.yaml)). To implement this, we need to introduce a YAML file in the [ConfigMap](https://github.com/longhorn/longhorn/blob/master/chart/templates/default-setting.yaml#L8) or a new ConfigMap YAML file.

In summary, we need to:

- Update chart/values.yaml
- Update chart/templates/default-setting.yaml (if introduce YAML file in the  [ConfigMap](https://github.com/longhorn/longhorn/blob/master/chart/templates/default-setting.yaml#L8)) or a new template ConfigMap YAML file.
- Regenerate new manifests
- Modify the Longhorn Manager's settings parsing code.

Unfortunately, the changes are unable to be applied in v1.8.0.-rc2. It is not a big change, so let's add in v1.8.0-rc3 ([release date will be 1/3](https://github.com/longhorn/longhorn/milestone/52)).

cc @ja@innobead

**Comment by derekbit** [comment]
@mantissahz 
We can move the three settings from [chart/values.yaml#L196](https://github.com/longhorn/longhorn/blob/master/chart/values.yaml#L196) into **a new category**, such as default-backupstore. However, the rendered manifest generated from the Helm values will still be included in default-settings.yaml ([source code](https://github.com/longhorn/longhorn/blob/master/chart/templates/default-setting.yaml)). To implement this, we need to introduce a YAML file in the [ConfigMap](https://github.com/longhorn/longhorn/blob/master/chart/templates/default-setting.yaml#L8) or a new ConfigMap YAML file.

In summary, we need to:

- Update chart/values.yaml
- Update chart/templates/default-setting.yaml (if introduce YAML file in the  [ConfigMap](https://github.com/longhorn/longhorn/blob/master/chart/templates/default-setting.yaml#L8)) or a new template ConfigMap YAML file.
- Regenerate new manifests
- Modify the Longhorn Manager's settings parsing code.

Unfortunately, the changes are unable to be applied in v1.8.0.-rc2. It is not a big change, so let's add in v1.8.0-rc3 ([release date will be 1/3](https://github.com/longhorn/longhorn/milestone/52)).

cc @innobead

**Comment by derekbit** [comment]
The doc and important note can be addressed in https://github.com/longhorn/longhorn/issues/10043

**Comment by roger-ryao** [comment]
> The doc and important note can be addressed in [#10043](https://github.com/longhorn/longhorn/issues/10043)

Removed the `require/doc` label as the documentation will be addressed in [#10043](https://github.com/longhorn/longhorn/issues/10043).

**Comment by roger-ryao** [comment]
**Verified on master-head 20250105**

- longhorn `master-head` https://github.com/longhorn/longhorn/commit/3f5aa24097b900029871b86509192afcb84ec06c
- longhorn-manager `master-head` https://github.com/longhorn/longhorn-manager/commit/8ab4efe7c77e56e352a6b8ecc17dde13da60a085

**Verified on v1.8.x-head 20250105**

- longhorn `v1.8.x-head` https://github.com/longhorn/longhorn/commit/3d69f6a2b31500d0021b4bce3a42b08331df3297
- longhorn-manager `v1.8.x-head` https://github.com/longhorn/longhorn-manager/commit/0b5ef0c67f3fb4f881c4ec881e9f351bdd96fbe1


**The test steps**
https://github.com/longhorn/longhorn/issues/10089#issuecomment-2565134425

**Result Passed**

- [x] Fresh installation of `v1.8.x-head` using Helm.
- [x] Upgrade from `v1.7.2` to `v1.8.x` using Helm.
- [x] Installation of `master-head` using Helm.
